src/bootstrap/toolstate.rs

```diff
@@ -1,4 +1,21 @@
 use serde::{Deserialize, Serialize};
+use build_helper::t;
+use std::time;
+use std::fs;
+use std::io::{Seek, SeekFrom};
+use std::collections::HashMap;
+use crate::builder::{Builder, RunConfig, ShouldRun, Step};
+use std::fmt;
+use std::process::Command;
+use std::path::PathBuf;
+use std::env;
+
+#[cfg(unix)]
+const OS: &str = "linux";
+#[cfg(not(unix))]
+const OS: &str = "windows";
```

Comment:
Can we have this? I don't want macOS to be treated as Linux if we (accidentally or not) run toolstate on it.

```rust
#[cfg(linux)]
const OS: Option<&str> = Some("linux");

#[cfg(windows)]
const OS: Option<&str> = Some("windows");

#[cfg(all(not(linux), not(windows)))]
const OS: Option<&str> = None;
```